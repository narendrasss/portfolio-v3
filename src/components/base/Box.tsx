import styled from 'styled-components';
import {
  style,
  space,
  width,
  color,
  flex,
  order,
  position,
  top,
  left,
  bottom,
  right,
  alignSelf,
  fontSize,
  FontSizeProps,
  SpaceProps,
  WidthProps,
  ColorProps,
  FlexProps,
  OrderProps,
  AlignSelfProps,
  PositionProps,
  TopProps,
  LeftProps,
  BottomProps,
  RightProps,
} from 'styled-system';

export interface BoxProps
  extends SpaceProps,
    WidthProps,
    ColorProps,
    FlexProps,
    OrderProps,
    AlignSelfProps,
    FontSizeProps,
    PositionProps,
    TopProps,
    LeftProps,
    BottomProps,
    RightProps {
  height?: string;
  display?: string | number[];
  testID?: string;
}

const height = style({
  key: 'heights',
  prop: 'height',
});

const display = style({
  key: 'displays',
  prop: 'display',
});

const Box = styled.div<BoxProps>`
  ${space}
  ${width}
  ${color}
  ${flex}
  ${order}
  ${alignSelf}
  ${fontSize}
  ${height}
  ${display}
  ${position}
  ${top}
  ${left}
  ${bottom}
  ${right}
`;

export default Box;
