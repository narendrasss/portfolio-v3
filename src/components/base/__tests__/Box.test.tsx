import React from 'react';
import { render } from 'react-testing-library';
import Box from '../Box';

describe('Box', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
  });
});
