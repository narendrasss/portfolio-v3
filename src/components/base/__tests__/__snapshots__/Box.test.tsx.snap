// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa cTECYE"
  />
</DocumentFragment>
`;
